src/Sarif/PropertyBag.cs

```diff
@@ -16,7 +16,12 @@ public class PropertyBagDictionary : TypedPropertyBagDictionary<object>
     {
         internal const string DEFAULT_POLICY_NAME = "default";
 
-        public PropertyBagDictionary() : base() { }
+        public PropertyBagDictionary() : this(null) { }
+
+        public PropertyBagDictionary(PropertyBagDictionary initializer) :
+            this(initializer, null)
+        {
+        }
 
         public PropertyBagDictionary(
             PropertyBagDictionary initializer = null,
```

Comment:
Oh right, one more thing about this. If you have an constructor that takes no arguments, and a constructor that takes two arguments, _both of which have defaults_, and you write:

```
X x = new X();
```

... then the call is ambiguous. The compiler didn't know if you meant to call the no-argument constructor, or to call the two-argument constructor with the default arguments. I don't think the code as currently written will build. Did you try building it?

---
In reply to: [66078330](https://github.com/Microsoft/sarif-sdk/pull/338#discussion_r66078330) [](ancestors = 66078330)